### PR TITLE
DW-2263 - Healthcheck endpoint for data-key-service and tests.

### PR DESCRIPTION
**Pull request by DanielChicot**
New GET endpoint on ‘/healthcheck’ reports the following

- all dependencies can be reached,
- the current master key id can be fetched,
- a new key can be generated,
- the new key is encrypted
- the new key can be decrypted

Response code is 200 if everything is OK, 500 otherwise. Example response body:

{
  "encryptionService": "OK",
  "masterKey": "OK",
  "dataKeyGenerator": "OK",
  "encryption": "OK",
  "decryption": “BAD”
}

### CODE REVIEW

File: src/test/java/uk/gov/dwp/dataworks/integration/HealthCheckIntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
+package uk.gov.dwp.dataworks.integration;
```
Comment (mark-r-m) [inline src/test/java/uk/gov/dwp/dataworks/integration/HealthCheckIntegrationTests.java]: Would you say these are Integration tests or Unit? We have Mocks here, so is it a Unit test (of a module)?

Comment (DanielChicot) [reply src/test/java/uk/gov/dwp/dataworks/integration/HealthCheckIntegrationTests.java]: The rationale was that if I was hitting the service endpoints, i.e. the http endpoints exposed by the service with the mockmvc controller, then this is classed as an integration test as the entity under test is not an individual function but an entire facet of the service as a whole.  The unit test package has tests for the functions of the individual classes/providers.

Comment (mark-r-m) [reply src/test/java/uk/gov/dwp/dataworks/integration/HealthCheckIntegrationTests.java]: Sounds good to me :-)

File: src/test/java/uk/gov/dwp/dataworks/integration/DataKeyIntegrationTests.java [hunk 2]
```diff
@@ -130,4 +130,5 @@ public void shouldReturnBadRequestWhenEncryptedDataKeyTooLarge() throws Exceptio
                 .andExpect(status().isBadRequest());
     }
 
+
```
Comment (mark-r-m) [inline src/test/java/uk/gov/dwp/dataworks/integration/DataKeyIntegrationTests.java]: Whitespace

**Pull request merged**